components/scream/src/dynamics/homme/physics_dynamics_remapper.cpp

```diff
@@ -412,10 +412,18 @@ do_remap_fwd() const
 #else
   const int team_size = std::min(1024, std::min(128*m_num_phys_cols,32*(concurrency/this->m_num_fields+31)/32));
 #endif
-#else
+#endif
+
+#ifdef KOKKOS_ENABLE_HIP
+  const int team_size = std::min(256, std::min(128*m_num_phys_cols,32*(concurrency/this->m_num_fields+31)/32));
```

Comment:
This line will need attn of @bartgol  or @ambrad .

Reply:
Does HIP choke on a team_size larger than 256 in non-debug builds? For CUDA, we need 256 only for debug...

If HIP can handle 1024 in non-debug builds, we could merge this ifdef with the one above, and use `#ifdef HOMMEXX_ENABLE_GPU` instead.

Reply:
I agree with Luca. I thought HIP has no issues with 1024 threads/team, so the original block, modified to use EAMXX_ENABLE_GPU, ought to work.

@bartgol, that aside, would you explain `32*(concurrency/this->m_num_fields+31)/32` to me? Doesn't order of ops imply that will always evaluate to `concurrency/this->m_num_fields+31`? That aside, if physics_dynamics_remapper is heavily used during time stepping, we should revisit it, as the parallelism looks like it could be greatly improved. Am I correct that with a pg2 grid, it's not used during time stepping?

Reply:
> @bartgol, that aside, would you explain `32*(concurrency/this->m_num_fields+31)/32` to me? Doesn't order of ops imply that will always evaluate to `concurrency/this->m_num_fields+31`?
Uhm I think you are correct. We are missing a set of parentheses.

> That aside, if physics_dynamics_remapper is heavily used during time stepping, we should revisit it, as the parallelism looks like it could be greatly improved.
Perhaps, yes. I think the original impl was doing a serial for loop over fields, and doing outer pfor over cols, and inner pfor over levs (for 3d fields). @tcclevenger might recall some numbers for how much the newer impl affected performance. We can take this offline, if you have some thoughts. The only upgrade I can think of is to do an outer pfor over numFields*numCols, to avoid having a block doing a "cheap" 2d field while another does a large 3d field. But it might make impl a bit more involved.

> Am I correct that with a pg2 grid, it's not used during time stepping?
Yes.

Reply:
I don't remember off the top of my head (performance new vs old), but I should have the numbers in a spreadsheet somewhere.. I'll try and find that.

Reply:
> Am I correct that with a pg2 grid, it's not used during time stepping?
> Yes.

Then it's fine to leave this on the back burner since it's used only in init for the production config.